In [1]:
clear all

## install this package for multiple regresion comparison

In [2]:
ssc install estout, replace

checking estout consistency and verifying not already installed...
all files already exist and are up to date.


## load data

In [3]:
use "~/Data/Dropbox/Working_Directory/Tex_MD/Teaching/tutorial/6002/W4/CARD.DTA", clear

## (a) Estimate $$\log\left(wage\right) = \beta_0 + \beta_1 \cdot educ + \beta_2 \cdot exper + \beta_3 \cdot exper^2 + \beta_4 \cdot black + \beta_5 \cdot SMSA + \beta_6 \cdot south + u$$

In [4]:
reg lwage educ exper expersq black smsa south

estimate store M_OLS



      Source |       SS           df       MS      Number of obs   =     3,010
-------------+----------------------------------   F(6, 3003)      =    204.93
       Model |  172.165628         6  28.6942714   Prob > F        =    0.0000
    Residual |  420.476016     3,003  .140018653   R-squared       =    0.2905
-------------+----------------------------------   Adj R-squared   =    0.2891
       Total |  592.641645     3,009  .196956346   Root MSE        =    .37419

------------------------------------------------------------------------------
       lwage |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
        educ |    .074009   .0035054    21.11   0.000     .0671357    .0808823
       exper |   .0835958   .0066478    12.57   0.000     .0705612    .0966305
     expersq |  -.0022409   .0003178    -7.05   0.000    -.0028641   -.0016177
       black |  -.1896315   .0176266   -10.76   0

## (b) Estimate $$\log\left(wage\right) = \beta_0 + \beta_1 \cdot educ + \beta_2 \cdot exper + \beta_3 \cdot exper^2 + \beta_4 \cdot black + \beta_5 \cdot SMSA + \beta_6 \cdot south + u$$ by the OLS with IQ as the proxy to ability $(u)$. Compare the estimate of $\beta_1$ using the proxy variable method to the regular OLS estimate in (a); Can we conclude that the OLS (without proxy) overestimate/underestimate $\beta_1$? Explain briefly.

In [5]:
reg lwage educ exper expersq black smsa south IQ

estimate store M_Proxy



      Source |       SS           df       MS      Number of obs   =     2,061
-------------+----------------------------------   F(7, 2053)      =     85.49
       Model |  81.1691559         7  11.5955937   Prob > F        =    0.0000
    Residual |  278.452104     2,053  .135631809   R-squared       =    0.2257
-------------+----------------------------------   Adj R-squared   =    0.2231
       Total |   359.62126     2,060  .174573427   Root MSE        =    .36828

------------------------------------------------------------------------------
       lwage |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
        educ |   .0692646    .004872    14.22   0.000     .0597099    .0788192
       exper |   .0935215   .0095356     9.81   0.000     .0748211    .1122219
     expersq |  -.0026762   .0004916    -5.44   0.000    -.0036403    -.001712
       black |   -.136135   .0262767    -5.18   0

### Comparing the estimate of β1 using the proxy variable (PV) method to the regular OLS estimate, the PV estimate is slightly smaller than the OLS one but the difference is insignificant. We cannot conclude that the OLS (without proxy) overestimate/underestimate $\beta_1$ because the IQ score is available in about $2/3$ of observations in the sample which would lead to sample selection bias.

## (c) An alternative method is to use the instrumental variable (IV) method. In order for IV to be consistent, the IV for educ, nearc4, must be uncorrelated with u. Could nearc4 be correlated with things in the error term, such as unobserved ability? Explain briefly.

### Although ability is generally unobservable, IQ score would be the proxy to ability as in (b). Furthermore, IQ scores are known to vary by geographic region, and so does the availability of four year colleges. It could be that, for a variety of reasons, people with higher abilities grow up in areas with four year colleges nearby. This is an example of geographic clustering

## (d) In general, the exogeneity (validity) condition cannot be tested directly. With the proxy IQ, however, we may get some evidence for the validity of nearc4 . Regress $IQ$ on $nearc4$ to check whether average IQ scores vary by whether the man grew up near a four-year college, controlling for SMSA and South. Are $IQ$ and $nearc4$reg IQ nearc4 smsa south related after the geographic dummy variables have been controlled for?

In [6]:
reg IQ nearc4 smsa south


      Source |       SS           df       MS      Number of obs   =     2,061
-------------+----------------------------------   F(3, 2057)      =     29.79
       Model |   20401.674         3  6800.55801   Prob > F        =    0.0000
    Residual |  469656.378     2,057   228.32104   R-squared       =    0.0416
-------------+----------------------------------   Adj R-squared   =    0.0402
       Total |  490058.052     2,060  237.892258   Root MSE        =     15.11

------------------------------------------------------------------------------
          IQ |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      nearc4 |    .890227   .7821444     1.14   0.255    -.6436504    2.424105
        smsa |      1.629   .8034407     2.03   0.043     .0533579    3.204642
       south |  -5.991017   .7176663    -8.35   0.000    -7.398445   -4.583588
       _cons |   102.6549   .8374004   122.59   0.

### The relationship between IQ and nearc4 is statistically insignificant (with p-value at $0.255$). Thus, once we control for region and environment of growing up, there is no apparent link between IQ score and living near a four-year college. This shows the importance to include geographic dummy variables in the wage equation to control for differences in access to colleges that might also be correlated with ability.

## (e) Another condition for $nearc4$ to be an IV for $educ$ is relevance. Test this condition based on the regression of $educ$ on $nearc4$, along with $black$, $smsa$ and $south$.

In [7]:
reg educ nearc4 black smsa south


      Source |       SS           df       MS      Number of obs   =     3,010
-------------+----------------------------------   F(4, 3005)      =     96.94
       Model |   2464.4422         4  616.110551   Prob > F        =    0.0000
    Residual |  19097.6379     3,005  6.35528714   R-squared       =    0.1143
-------------+----------------------------------   Adj R-squared   =    0.1131
       Total |  21562.0801     3,009  7.16586243   Root MSE        =     2.521

------------------------------------------------------------------------------
        educ |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
      nearc4 |   .3317504   .1070669     3.10   0.002     .1218186    .5416822
       black |  -1.463773   .1155921   -12.66   0.000    -1.690421   -1.237126
        smsa |   .8477946   .1093822     7.75   0.000      .633323    1.062266
       south |   -.453838   .1026328    -4.42   0.

### This is the standard t test on the coefficient on nearc4. Form the t test statistic for the null; if t stat exceeds critical value (the p-value is smaller that the significance level, usually 5%), reject the null and conclude $educ$ and $nearc4$ are correlated so that relevance condition is satisfied. From the estimates below, we reject the null hypothesis that the coefficient on $nearc4$ is zero.

## (f) The third condition for nearc4 to be an IV for educ is exclusion. To see this, run a regression of log(wage) based on (2.1) with two additional regressors: nearc4 and IQ. Is the coefficient on nearc4 significant?

In [8]:
reg lwage educ exper expersq black smsa south IQ nearc4


      Source |       SS           df       MS      Number of obs   =     2,061
-------------+----------------------------------   F(8, 2052)      =     74.83
       Model |  81.2192036         8  10.1524005   Prob > F        =    0.0000
    Residual |  278.402056     2,052  .135673517   R-squared       =    0.2258
-------------+----------------------------------   Adj R-squared   =    0.2228
       Total |   359.62126     2,060  .174573427   Root MSE        =    .36834

------------------------------------------------------------------------------
       lwage |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
        educ |   .0690458   .0048861    14.13   0.000     .0594636     .078628
       exper |   .0935021   .0095371     9.80   0.000     .0747987    .1122055
     expersq |  -.0026801   .0004917    -5.45   0.000    -.0036445   -.0017157
       black |  -.1366962    .026297    -5.20   0.

### Since for the coefficient on nearc4, the p-value 0.544 is much higher than 5%, the coefficient on nearc4 is insignificant.

## (g) Estimate (2.1) by the IV method using nearc4 as the IV for educ, report the esti- mates in the table. How is the IV estimate of β1 different from the OLS estimate?

In [9]:
ivregress 2sls lwage exper expersq black smsa south (educ=nearc4)

estimate store M_IV



Instrumental variables (2SLS) regression          Number of obs   =      3,010
                                                  Wald chi2(6)    =     726.69
                                                  Prob > chi2     =     0.0000
                                                  R-squared       =     0.2252
                                                  Root MSE        =     .39058

------------------------------------------------------------------------------
       lwage |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
        educ |   .1322888    .049176     2.69   0.007     .0359057    .2286719
       exper |    .107498   .0212758     5.05   0.000     .0657981    .1491978
     expersq |  -.0022841   .0003337    -6.84   0.000    -.0029382   -.0016299
       black |  -.1308019   .0528108    -2.48   0.013    -.2343091   -.0272946
        smsa |   .1313237   .0300948     4.36   0

### It is interesting to see that the IV estimate of the coefficient on educ is larger than the OLS estimate. This is somehow contrary to the conventional belief that the OLS overestimates the return to education. As suggested in Card (1995), one possible explanation for the positive gap between IV and OLS estimates of the return to ed- ucation is that the latter are downward-biased by measurement error in schooling; An alternative possibility is that the "true" rate of return to education varies across the population, and that the increase in education associated with college proximity occurs for individuals with relatively high rates of return to schooling.

## (h) Use the Wu-Hausman test to see whether educ is endogenous; that is, determine if the difference between OLS and IV is statistically significant.

In [10]:
estat endogenous


  Tests of endogeneity
  Ho: variables are exogenous

  Durbin (score) chi2(1)          =  1.54235  (p = 0.2143)
  Wu-Hausman F(1,3002)            =  1.53904  (p = 0.2149)


## Extra: 2SLS with two instruments (nearc4 and nearc2) for educ

In [11]:
ivregress 2sls lwage exper expersq black smsa south (educ=nearc4 nearc2)

estimate store M_2SLS



Instrumental variables (2SLS) regression          Number of obs   =      3,010
                                                  Wald chi2(6)    =     663.32
                                                  Prob > chi2     =     0.0000
                                                  R-squared       =     0.1455
                                                  Root MSE        =     .41017

------------------------------------------------------------------------------
       lwage |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
        educ |   .1608487   .0485725     3.31   0.001     .0656484    .2560491
       exper |   .1192112   .0211532     5.64   0.000     .0777516    .1606708
     expersq |  -.0023052   .0003502    -6.58   0.000    -.0029917   -.0016188
       black |  -.1019726   .0525575    -1.94   0.052    -.2049833    .0010382
        smsa |   .1165736   .0302782     3.85   0

## Compare them all

In [17]:
estout M_OLS M_Proxy M_IV M_2SLS, ///
       cells(b(star fmt(3)) se(par fmt(3))) ///
       stats(r2 F chi2 N, labels(R-squared F-value Chi2-value "sample size")) ///
       legend


----------------------------------------------------------------------------
                    M_OLS         M_Proxy            M_IV          M_2SLS   
                     b/se            b/se            b/se            b/se   
----------------------------------------------------------------------------
educ                0.074***        0.069***        0.132**         0.161***
                  (0.004)         (0.005)         (0.049)         (0.049)   
exper               0.084***        0.094***        0.107***        0.119***
                  (0.007)         (0.010)         (0.021)         (0.021)   
expersq            -0.002***       -0.003***       -0.002***       -0.002***
                  (0.000)         (0.000)         (0.000)         (0.000)   
black              -0.190***       -0.136***       -0.131*         -0.102   
                  (0.018)         (0.026)         (0.053)         (0.053)   
smsa                0.161***        0.153***        0.131***        0.117**

In [18]:
!jupyter nbconvert --to html W4_Stata.ipynb


[NbConvertApp] Converting notebook W4_Stata.ipynb to html
[NbConvertApp] Writing 606703 bytes to W4_Stata.html
